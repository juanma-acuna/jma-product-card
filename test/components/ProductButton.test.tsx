import React from 'react';
import render from 'react-test-renderer';
import { ProductButtons } from '../../src/components';

describe('ProductButtons', () => {
  test('debe mostrar el componente sin parametros adicionales', () => {
    const wrapper = render.create(<ProductButtons />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostrar el componente correctamente con una clase custom', () => {
    const wrapper = render.create(<ProductButtons className="custom-class" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('debe mostrar el componente correctamente con una clase custom y un estilo custom', () => {
    const wrapper = render.create(
      <ProductButtons
        className="custom-class"
        style={{ backgroundColor: 'black' }}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
