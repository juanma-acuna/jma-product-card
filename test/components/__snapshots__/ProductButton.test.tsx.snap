// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductButtons debe mostrar el componente correctamente con una clase custom 1`] = `
<div
  className="buttonsContainer custom-class"
>
  <button
    className="buttonMinus"
    onClick={[Function]}
  >
    -
  </button>
  <div
    className="countLabel"
  />
  <button
    className="buttonAdd false"
    disabled={false}
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`ProductButtons debe mostrar el componente correctamente con una clase custom y un estilo custom 1`] = `
<div
  className="buttonsContainer custom-class"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <button
    className="buttonMinus"
    onClick={[Function]}
  >
    -
  </button>
  <div
    className="countLabel"
  />
  <button
    className="buttonAdd false"
    disabled={false}
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`ProductButtons debe mostrar el componente sin parametros adicionales 1`] = `
<div
  className="buttonsContainer undefined"
>
  <button
    className="buttonMinus"
    onClick={[Function]}
  >
    -
  </button>
  <div
    className="countLabel"
  />
  <button
    className="buttonAdd false"
    disabled={false}
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
